src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt

```diff
@@ -545,6 +579,100 @@ object ManagedIndexRunner : ScheduledJobRunner,
         return result
     }
 
+    // delete metadata in cluster state
+    private suspend fun deleteManagedIndexMetaData(managedIndexMetaData: ManagedIndexMetaData): Boolean {
+        var result = false
+        try {
+            val request = UpdateManagedIndexMetaDataRequest(
+                indicesToRemoveManagedIndexMetaDataFrom = listOf(Index(managedIndexMetaData.index, managedIndexMetaData.indexUuid))
+            )
+            updateMetaDataRetryPolicy.retry(logger) {
+                val response: AcknowledgedResponse = client.suspendUntil { execute(UpdateManagedIndexMetaDataAction.INSTANCE, request, it) }
+                if (response.isAcknowledged) {
+                    result = true
+                } else {
+                    logger.error("Failed to delete ManagedIndexMetaData for [index=${managedIndexMetaData.index}]")
+                }
+            }
+        } catch (e: ClusterBlockException) {
+            logger.error("There was ClusterBlockException trying to delete the metadata for ${managedIndexMetaData.index}. Message: ${e.message}", e)
+        } catch (e: Exception) {
+            logger.error("Failed to delete ManagedIndexMetaData for [index=${managedIndexMetaData.index}]", e)
+        }
+        return result
+    }
+
+    /**
+     * update metadata in config index, and save metadata in history after update
+     * this can be called 2 times in one job run, so need to save seqNo & primeTerm
+     */
+    private suspend fun updateManagedIndexMetaData(
+        managedIndexMetaData: ManagedIndexMetaData,
+        lastUpdateResult: UpdateMetadataResult? = null
+    ): UpdateMetadataResult {
+        val result = UpdateMetadataResult()
+
+        var metadata: ManagedIndexMetaData = managedIndexMetaData
+        if (lastUpdateResult != null) {
+            metadata = managedIndexMetaData.copy(seqNo = lastUpdateResult.seqNo, primaryTerm = lastUpdateResult.primaryTerm)
+        }
+
+        val indexRequest = managedIndexMetadataIndexRequest(metadata)
+        try {
+            updateMetaDataRetryPolicy.retry(logger) {
+                val indexResponse: IndexResponse = client.suspendUntil { index(indexRequest, it) }
+                result.savedMetadata = indexResponse.status() == RestStatus.OK ||
```

Comment:
I'm assuming one of these is for an update and other is for initial create?

Reply:
yes